### PR TITLE
The Hypermatter state - a return

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Re-PR of #67112

Adds a new way to achieve power with the SM, the Hypermatter state.

It's a state where the crystal turns blue and all gas interactions will stop, even the damage will stop, what starts to happen is the production of nuclear particles shoot around the crystal. These particles have internal power than can be increased by using reflectors made from plastitanium glass and if redirect to the crystal after several reflections it can extend the duration of the hypermatter state. The particles can be also sent to a special nuclear accumulator that will convert their power into usable electricity.
To achieve the hyperstate you'll first need to feed the SM plasma coated metallic hydrogen (which is metallic hydrogen exposed to >2000 K plasma). After that the SM will enter the hyperstate, each sheet will increase the duration by 30 seconds to a max of 5 minutes (so 10 sheets needed, all the successive sheets will just be consumed).
The other way is to use a nuclear emitter that can be built from parts found in the engineering secure storage. This machine needs to accumulate power before it can emit the beam needed to activate the state.

the Hypermatter state also work on the shards, but they need to be anchored and stay anchored

Updated image with new visuals
image
The video still has the old visuals
https://user-images.githubusercontent.com/42839747/203993454-2ad6a988-c54e-41f2-8cd7-b60542b904f4.mp4
## Why It's Good For The Game

Expands on the experimental tone that the SM has by giving it a state where the crystal changes color and start releasing rad balls, achieved by feeding it a modified version of metallic hydrogen.
It's also a new way for engineers to generate power without the need of atmos since this state doesn't require any gas and doesn't produce any of them.
## Changelog
:cl:
add: hypermatter state, a new state of existence of the SM, where the crystal changes into a blue shade and shoots a great amount of rad balls, those can now be reflected on reflectors, plus there is a new type of one sided reflector made with plastitanium glass that can increase their internal power, you can now collect those inside a nuclear accumulator to generate power
add: hypermatter state can be achieved by shooting the crystal with a nuclear emitter.
add: shards can achieve this state but those need to be anchored and stay anchored to work
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
But do we have a use for this burst of energy production

**Comment by Jackal-boop** [comment]
this could probably have a fun interaction with cascades

**Comment by Wallemations** [comment]
As it stands I don't really see a use for this, but if we do that power bar thing mothblocks has been talking about this could be a really cool way to get additional bars of power.

**Comment by Ghilker** [comment]
> As it stands I don't really see a use for this, but if we do that power bar thing mothblocks has been talking about this could be a really cool way to get additional bars of power.

that's the plan yes

File: code/game/objects/structures/reflector.dm [hunk 1]
```diff
@@ -5,17 +5,33 @@
...
+	var/frame_build_stack_amount = 5
+	///The sheet types of the final build that are dropped from disassembling the reflector
+	var/build_stack_type = /obj/item/stack/sheet/iron
+	///Amount of sheets needed to build the reflector.
+	var/build_stack_amount = 0
+	///Typecache of the projectiles allowed to be reflected
+	var/list/allowed_projectile_typecache = list(
+		/obj/projectile/beam,
+		/obj/projectile/energy/nuclear_particle,
+		)
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/game/objects/structures/reflector.dm:30]: it actually is a typecache, if you look at line 39 you can see 
`allowed_projectile_typecache = typecacheof(allowed_projectile_typecache)`

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -589,6 +589,8 @@
 		"anomaly_refinery",
 		"tank_compressor",
 		"atmos_thermal",
+		"nuclear_accumulator",
+		"nuclear_emitter",
```
[the remark this answers is not in the document]
Comment (Ghilker) [reply code/modules/research/techweb/all_nodes.dm:593]: i'll sort it out completely

**Comment by LemonInTheDark** [comment]
Reposting from discord
This seems like you chafing under the restriction of only one mainline engine. Having to balance more then one "type" of supermatter at once seems like it just brings us back to the singulo/tesla days, and if so why did we even get rid of those.
I'd prefer you make the SM balanced, or something balanced, and then we can start adding other engines in.

will close soon unless overriden by someone

**Comment by Ghilker** [comment]
fine by me, i'll try to do the power rework first, then focus on the SM

**Pull request closed without merging**